jupyter_sphinx/main.py

```diff
@@ -0,0 +1,130 @@
+import pkg_resources
+from typing import List, Optional, Set
+
+from sphinx.application import Sphinx
+from sphinx.environment import BuildEnvironment
+from sphinx.errors import ExtensionError
+
+from jupyter_db import JupyterDB
+from .components import (
+    JupyterExec,
+    JupyterExecNode,
+    JupyterKernel,
+    JupyterView,
+    JupyterViewNode,
+    ResolveJupyterViews,
+)
+
+
+def setup(app: Sphinx) -> dict:
+    """Initialize Sphinx extension."""
+
+    app.add_config_value("jupyter_sphinx_executor", "default", "html")
+    app.add_config_value(
+        "jupyter_execute_data_priority",
+        [
+            "application/vnd.jupyter.widget-view+json",
+            "application/javascript",
+            "text/html",
+            "image/svg+xml",
+            "image/png",
+            "image/jpeg",
+            "text/latex",
+            "text/plain",
+        ],
+        "env",
+    )
+
+    # Document Elements
+    app.add_directive("jupyter-kernel", JupyterKernel)
+    app.add_directive("jupyter-exec", JupyterExec)
```

Comment:
Another question: is there a reason why you use `jupyter-exec`, and not `jupyter-execute`? I personally don't have a preference either way, but if possible I'd like to avoid changing conventions. Or is there another package that introduces a `jupyter-exec` directive?

Reply:
No it was absolutely an arbitrary choice, and can be changed to `jupyter-execute` 👍